commit 50695333f9ca2658d99a0da3eed737de953622f9 (refs/changes/48/2348/8)
Author: Tim Kordas <tim.kordas@joyent.com>
Date:   2017-09-13T23:43:08+00:00 (2 years, 1 month ago)
    
    MANTA-3311 mako_gc carefully checks lock file but barrels on if it can't create one
    Reviewed by: David Pacheco <dap@joyent.com>

diff --git a/bin/mako_gc.sh b/bin/mako_gc.sh
index 3e4dfa6..5c7ee34 100755
--- a/bin/mako_gc.sh
+++ b/bin/mako_gc.sh
@@ -196,7 +196,21 @@ if [[ -n "$LAST_PID" ]]; then
     fi
 fi
 
+# Save our noclobber setting
+ALLOWEDCLOBBER=1
+if [[ -o noclobber ]]; then
+    ALLOWEDCLOBBER=0
+fi
+set -o noclobber
 echo -n $PID >$PID_FILE
+if [[ $? -ne 0 ]]
+then
+    fatal "Failed to acquire lockfile"
+fi
+# Restore our saved noclobber setting
+if [[ $ALLOWEDCLOBBER -ne 0 ]]; then
+    set +o noclobber
+fi
 
 # Ok, we're good to start gc
 log "starting gc"
